mlir/lib/Dialect/SCF/Transforms/TileUsingInterface.cpp

```diff
@@ -679,6 +863,276 @@ mlir::scf::tileUsingSCF(RewriterBase &rewriter, TilingInterface op,
+  size_t numLoops = iterationDomain.size();
+  if (numLoops == 0) {
+    return rewriter.notifyMatchFailure(
+        op, "unable to tile op with no iteration domain");
+  }
+  // 2. Materialize the tile sizes. Enforce the convention that "tiling by zero"
+  // skips tiling a particular dimension. This convention is significantly
+  // simpler to handle instead of adjusting affine maps to account for missing
+  // dimensions.
+  SmallVector<OpFoldResult> tileSizeVector =
+      options.tileSizeComputationFunction(rewriter, op);
+  if (tileSizeVector.size() < iterationDomain.size()) {
+    auto zero = rewriter.getIndexAttr(0);
+    tileSizeVector.append(numLoops - tileSizeVector.size(), zero);
+  }
+
+  // 3. Find the destination tensors to use for the operation.
+  SmallVector<Value> destinationTensors;
+  if (failed(tensor::getOrCreateDestinations(rewriter, op.getLoc(), op,
+                                             destinationTensors))) {
+    return rewriter.notifyMatchFailure(op,
+                                       "unable to create destination tensors");
+  }
+
+  SmallVector<OpFoldResult> offsets, sizes;
+  SmallVector<LoopLikeOpInterface> forLoops;
+  // Several innermost loops are created when applying continuous tiling.
+  // All innermost loops need to be processed in an identical manner.
+  SmallVector<scf::ForOp> innermostLoops;
+  {
+    // If there is an interchange specified, permute the iteration domain and
+    // the tile sizes.
+    SmallVector<int64_t> interchangeVector;
+    if (!options.interchangeVector.empty()) {
+      interchangeVector = fillInterchangeVector(options.interchangeVector,
+                                                iterationDomain.size());
+
+      if (!isPermutationVector(interchangeVector)) {
+        return rewriter.notifyMatchFailure(
+            op, "invalid intechange vector, not a permutation of the entire "
+                "iteration space");
+      }
+
+      applyPermutationToVector(iterationDomain, interchangeVector);
+      applyPermutationToVector(tileSizeVector, interchangeVector);
+    }
+
+    // Specify which loop nesting should continuously tiled.
+    SmallVector<bool> cTileVector = options.continuousTileMappingVector;
+
+    if (cTileVector.size() < iterationDomain.size())
+      cTileVector.append(numLoops - cTileVector.size(), false);
+
+    offsets.resize(iterationDomain.size());
+    sizes.resize(iterationDomain.size());
+
+    // 4. Materialize an empty loop nest that iterates over the tiles. These
+    // loops for now do not return any values even if the original operation has
+    // results.
+    forLoops = generateContinuousTileLoopNest(
+        rewriter, op.getLoc(), iterationDomain, tileSizeVector, cTileVector,
+        sizesMap, innermostLoops, destinationTensors);
+  }
+
+  LLVM_DEBUG({
+    if (!forLoops.empty()) {
+      llvm::dbgs() << "LoopNest shell :\n";
+      forLoops.front()->getBlock()->dump();
```

Comment:
You likely want to `->print(llvm::dbgs())` for the case when error stream and debug stream differ.